internal/controller/api/v1alpha1/nmstate_config_types.go

```diff
@@ -20,19 +20,19 @@ import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
-type RawConfig []byte
+type MacInterfaceMapItems struct {
+	// nic name used in the yaml, which relates 1:1 to the mac address
+	LogicalNicName string `json:"logicalNICName"`
+	// mac address present on the host.
+	// Pattern: ^([0-9A-Fa-f]{2}[:]){5}([0-9A-Fa-f]{2})$
+	MacAddress string `json:"macAddress"`
+}
 
 type NMStateConfigSpec struct {
-	// MACAddress is a MAC address for any network device on the host to which this config
-	// should be applied. This value is only used to ensure that the config is applied to the
-	// intended host.
-	MACAddress string `json:"macAddress"`
-
-	//Config contains the  yaml [1] as string instead of golang struct so we don't need to be in
-	//sync with the schema.
-	//
-	// [1] https://github.com/nmstate/nmstate/blob/base/libnmstate/schemas/operational-state.yaml
-	Config RawConfig `json:"-"`
+	// mapping of host macs to logical interfaces used in the network yaml
+	MacInterfaceMap []*MacInterfaceMapItems `json:"macInterfaceMap,omitempty"`
+	// yaml string that can be processed by nmstate
+	NetworkYaml string `json:"networkYAML,omitempty"`
```

Comment:
We want the experience of expressing nmstate config to be similar to that of the `NodeNetworkConfigurationPolicy`, where the nmstate yaml is basically in-lined as part of the Spec. See the [example in their README](https://github.com/nmstate/kubernetes-nmstate/blob/master/README.md).

You can see some details on how they're accomplishing that in [their code](https://github.com/nmstate/kubernetes-nmstate/blob/master/api/shared/state_types.go). Making this field a `string` type would be a different experience.

Personally I prefer the simpler field name "config". Forcing the content to be in-line will make it implicitly YAML, so a more specific noun is warranted. And I think "network" is implied since the resource is called `NMStateConfig`. But I'm open to other opinions. Naming is hard.

Reply:
Done